src/test/scala/org/encryfoundation/prismlang/evaluator/EvaluatorSpec.scala

```diff
@@ -162,6 +162,74 @@ class EvaluatorSpec extends PropSpec with Matchers with TestCompiler with ExprEv
     resultTry.get shouldBe List(2, 4, 6)
   }
 
+  property("Filter") {
+
+    val expr: Expr = Call(
+      Attribute(
+        Collection(
+          List(IntConst(1), IntConst(2), IntConst(3), IntConst(4), IntConst(5)),
+          Types.Nit
+        ),
+        Ident("filter"),
+        Types.Nit
+      ),
+      List(
+        Lambda(
+          List(
+            (Ident("i"), TypeIdent("Int", List()))
+          ),
+          Compare(
+            Name(Ident("i"), Types.Nit),
+            List(CompOp.Gt),
+            List(IntConst(3))
+          ),
+          Types.Nit
+        )
+      ),
+      Types.Nit
+    )
+
+    val resultTry: Try[Any] = eval(compileExpr(expr).get)
+
+    resultTry.isSuccess shouldBe true
+
+    resultTry.get shouldBe List(4, 5)
+  }
+
+  property("Filter (empty result)") {
+
+    val expr: Expr = Call(
+      Attribute(
+        Collection(
+          List(IntConst(1), IntConst(2)),
+          Types.Nit
+        ),
+        Ident("filter"),
+        Types.Nit
+      ),
+      List(
+        Lambda(
+          List(
+            (Ident("i"), TypeIdent("Int", List()))
+          ),
+          Compare(
+            Name(Ident("i"), Types.Nit),
+            List(CompOp.Gt),
+            List(IntConst(3))
+          ),
+          Types.Nit
+        )
+      ),
+      Types.Nit
+    )
+
+    val resultTry: Try[Any] = eval(compileExpr(expr).get)
+
+    resultTry.isSuccess shouldBe true
+
+    resultTry.get shouldBe List()
```

Comment:
it's better to use `org.scalatest.TryValues` which allows to write `resultTry.value shouldBe List()` in exception safe manner, then checking `isSuccess` will be redundant